base/tuple.jl

```diff
@@ -352,3 +352,72 @@ any(x::Tuple{Bool, Bool, Bool}) = x[1]|x[2]|x[3]
 Returns an empty tuple, `()`.
 """
 empty(x::Tuple) = ()
+
+## Linked-list representation of a tuple. Inferrable even for Type elements.
```

Comment:
I'm not quite sure what the purpose of the changes to this type are about. It's not fully-inferable for Type elements (to some extent, that would require the type system to be fixed, but beyond that it's also just a meaningless statement). Unlike Tuple, this type is not intended for general use. It's very specific to `broadcast`, which is why everything was originally unrolled (e.g. assuming a small number of arguments) and tucked away in `Base.Broadcast` (we don't need a copy of this in Core.Inference). What's the purpose of the special type `TupleLLEnd` also (I originally used `Nothing` for this)?

Reply:
We could move it back to Broadcast if you prefer. I wondered if it might be more generally useful, esp. since I thought ([incorrectly, it seems](https://github.com/JuliaLang/julia/pull/25377#discussion_r160313681)) that it might help solve the O(N^2) compilation problem.

With regards to TupleLLEnd, see https://github.com/JuliaLang/julia/pull/25377#discussion_r159956996